articles/quickstart/spa/angular-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,459 @@
+<!-- markdownlint-disable MD041 MD034 MD002 -->
+
+<%= include('../../_includes/_login_preamble', { library: 'Angular 7+' }) %>
+
+This tutorial will guide you in modifying an Angular application to support login using Auth0, that demonstrates how users can log in, log out, and how to view their profile information. It will also show you how to protect routes from unauthenticated users.
+
+::: note
+This tutorial assumes that you're able to use the [Angular CLI](https://cli.angular.io/) to generate new components and services on your project. If not, please place the files where appropriate and ensure that any references to them regarding import paths are correct for your scenario.
+:::
+
+## Create an Angular application
+
+If you do not already have an application in which to implement the SDK, a new one can be generated using the [Angular CLI](https://cli.angular.io/).
+
+Install the CLI:
+
+```bash
+npm install -g @angular/cli
+```
+
+Then use it to generate a new Angular application. When asked _"Would you like to use Angular routing?"_, select **yes**:
+
+```bash
+ng new auth0-demo
+```
+
+Once the project has been created, open the `auth0-demo` folder in your favorite code editor.
+
+## Install the SDK
+
+While in the project folder, install the SDK using `npm` in the terminal:
+
+```bash
+npm install @auth0/auth0-spa-js
+```
+
+## Add an Authentication Service
+
+To make it easier to use the Auth0 SDK within an Angular application, create an Angular service that can be injected into other components and services. This service can be used to set up and retrieve an instance of `Auth0Client`.
+
+To start, generate a new service called `Auth`:
+
+```bash
+ng generate service Auth
+```
+
+Open the `src/app/auth.service.ts` file inside your code editor and add the following content:
+
+```ts
+// src/app/auth.service.ts
+
+import { Injectable } from '@angular/core';
+import createAuth0Client from '@auth0/auth0-spa-js';
+import Auth0Client from '@auth0/auth0-spa-js/dist/typings/Auth0Client';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  isAuthenticated = new BehaviorSubject(false);
+  profile = new BehaviorSubject<any>(null);
+
+  private auth0Client: Auth0Client;
+
+  // Auth0 application configuration
+  config = {
+    domain: "${account.tenant}",
+    client_id: "${account.clientId}"
+  };
+
+  /**
+   * Gets the Auth0Client instance.
+   */
+  async getAuth0Client(): Promise<Auth0Client> {
+    if (!this.auth0Client) {
```

Comment:
Maybe switch this to check `if (this.auth0Client) {` and return `this.auth0Client` early?

Reply:
@joshcanhelp Will update this in the sample app as well